### PR TITLE
Update Beginner-guide

### PR DESCRIPTION
**Pull request by puneeth072003**
#### Adding contents to CONTRIBUTING.md and Renaming the Images as Requested in #586 working on the Issue #558

### CODE REVIEW

**Comment by puneeth072003** [comment]
@commjoen  @bendehaan Anything wrong?

**Comment by commjoen** [comment]
> @commjoen @bendehaan Anything wrong?
> this test fails every time for me

Hi @puneeth072003 : it looks like it does not like the formatting…

**Comment by puneeth072003** [comment]
@commjoen how can i fix it?

**Comment by commjoen** [comment]
https://stackoverflow.com/questions/10418975/how-to-change-line-ending-settings can help I think

**Comment by puneeth072003** [comment]
@bendehaan @commjoen Anything left out?

**Comment by bendehaan** [comment]
Great progress! Before we can get this merged I would like to see a practical example of how to alter or add to something in the application as well. Could you add that?

**Comment by puneeth072003** [comment]
I think that it would take a while.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -67,18 +73,15 @@ Pull requests should be as small/atomic as possible. Large, wide-sweeping change
...
 
 5. Choose what to work on, based on any of the outstanding [issues](https://github.com/OWASP/wrongsecrets/issues "WrongSecrets Issues").
-6. Create a branch so that you can cleanly work on the chosen issue: `git checkout -b fix/Issue66`
+6. Create a branch so that you can cleanly work on the chosen issue: `git checkout -b FixingIssue66`
 7. Open your favorite editor and start making modifications. We recommend using the [IntelliJ Idea](https://www.jetbrains.com/idea/).
-8. Install [pre-commit](https://pre-commit.com/#install) the dependencies for our pre-commit configuration to make sure your code complies with standards used in the project. This requires terraform, [terraform-docs](https://github.com/terraform-docs/terraform-docs#installation), [tflint](https://github.com/terraform-linters/tflint#installation), and [commitlint](https://commitlint.js.org/#/guides-local-setup). For commitlint, you need [NodeJS](https://nodejs.org/en/download/) installed, after which you you can use `npm install` in the root folder of this project.
```
Comment (commjoen) [inline CONTRIBUTING.md:75]: Looks like we had some modificatios of the document while you were working on the doc. Can you please put this back?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -67,18 +73,15 @@ Pull requests should be as small/atomic as possible. Large, wide-sweeping change
...
-9. Install the pre-commit hook using `pre-commit install --hook-type commit-msg`. We recommend to run `pre-commit run -a` every so often if you're working on a bigger change.
```
Comment (commjoen) [inline CONTRIBUTING.md:76]: same here, can you please put this back?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -67,18 +73,15 @@ Pull requests should be as small/atomic as possible. Large, wide-sweeping change
...
-10. After your modifications are done, push them to your forked repository. This can be done by executing the command `git add MYFILE` for every file you have modified, followed by `git commit -m 'your commit message here'` to commit the modifications and `git push` to push your modifications to GitHub.
-11. Create a Pull Request (PR) by going to your fork, <https://github.com/Your_Github_Handle/wrongsecrets> and click on the "New Pull Request" button. The target branch should typically be the Master branch. When submitting a PR, be sure to follow the checklist that is provided in the PR template. The checklist itself will be filled out by the reviewer.
-12. If something in your git workflow went wrong (and e.g., the precommit hook CI run failed), check out ["O Shit, Git!?!"](https://ohshitgit.com/) to view tips on editing your historical commit message(s), among others.
```
Comment (commjoen) [inline CONTRIBUTING.md:79]: same here, can you please put this back?

File: scripts/get_helm.sh [hunk 4]
```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/env bash
```
Comment (commjoen) [inline scripts/get_helm.sh:1]: I think we should do without the scripts folder, can you please remove it :) ?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -99,3 +102,141 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+      ![](images/final-output-8.png)
+
+    - Here is the preview of the **web server**,  you can try to find the secrets by means of solving the challenge offered at:
+      [**Challenges**](https://github.com/OWASP/wrongsecrets#basic-docker-exercises)
+        ![](images/screenshot.png)
+
+---
+## How to add challenge
+- ### Step 1: Creating a new issue.
+  First make sure that you have an [Issue](https://github.com/OWASP/wrongsecrets/issues/new) reported for which a challenge is really wanted.
```
Comment (commjoen) [inline CONTRIBUTING.md:214]:
```suggestion
  First make sure that you have an [Issue](https://github.com/OWASP/wrongsecrets/issues/new) reported for which a challenge is really wanted.
  And make sure the challenge is assigned to you, as others might be working at the challenge.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -99,3 +102,141 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+
+- ### Step 2: Adding the challenge.
+  Add the **new challenge** in this folder `wrongsecrets/src/main/java/org/owasp/wrongsecrets/challenges/`.
+  These are the things that you have to keep in mind.
+    - First and foremost make sure your challenge is coded in **Java**.
+    - Don't forget to add your challenge number in `@Order(28)` annotation, ***28*** in my case.
+    - For further reference please refer [*Challenge28.java*](https://github.com/puneeth072003/wrongsecrets/blob/Readme-updation/src/main/java/org/owasp/wrongsecrets/challenges/docker/Challlenge28.java)
```
Comment (commjoen) [inline CONTRIBUTING.md:221]:
```suggestion
Here is an example of a possible Challenge 28:
```java
package org.owasp.wrongsecrets.challenges.docker;


import lombok.extern.slf4j.Slf4j;
import org.owasp.wrongsecrets.RuntimeEnvironment;
import org.owasp.wrongsecrets.ScoreCard;
import org.owasp.wrongsecrets.challenges.Challenge;
import org.owasp.wrongsecrets.challenges.ChallengeTechnology;
import org.owasp.wrongsecrets.challenges.Spoiler;
import org.springframework.core.annotation.Order;
import org.springframework.stereotype.Component;

import java.util.List;

@Slf4j
@Component
@Order(28) //make sure this number is the same as your challenge
public class Challenge28 extends Challenge {
    private final String secret;

    public Challenge28(ScoreCard scoreCard) {
        super(scoreCard);
        secret = "hello world";
    }

    //is this challenge usable in CTF mode?
    @Override
    public boolean canRunInCTFMode() {
        return true;
    }

    //return the plain text secret here
    @Override
    public Spoiler spoiler() {
        return new Spoiler(secret);
    }


    //here you validate if your answer matches the secret
    @Override
    public boolean answerCorrect(String answer) {
        return secret.equals(answer);
    }

    //which runtime can you use to run the challenge on ? (You can just use Docker here)
    @Override
    public List<RuntimeEnvironment.Environment> supportedRuntimeEnvironments() {
        return List.of(RuntimeEnvironment.Environment.DOCKER);
    }

    //set the difficulty: 1=low, 5=very hard
    @Override
    public int difficulty() {
        return 1;
    }

    //on which tech is this challenge? See ChallengeTechnology.Tech for categories
    @Override
    public String getTech() {
        return ChallengeTechnology.Tech.SECRETS.id;
    }

    //if you use this in a shared environment and need to adapt it, then return true here.
    @Override
    public boolean isLimittedWhenOnlineHosted() {
        return false;
    }

}
```
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -99,3 +102,141 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+
+- ### Step 3: Adding Test File.
+  Add the **new TestFile** in this folder `wrongsecrets/src/test/java/org/owasp/wrongsecrets/challenges/`. TestFile is required to do **unit testing.**
```
Comment (commjoen) [inline CONTRIBUTING.md:224]:
```suggestion
  Add the **new TestFile** in this folder `wrongsecrets/src/test/java/org/owasp/wrongsecrets/challenges/`. TestFile is required to do **unit testing.**
  Please note that PRs for new challenges are only excepted when unit tests are added to prove that the challenge works. Normally tests should not immediately leak the actual secret, so leverage the `.spoil()` functionality of your test implementation for this.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -99,3 +102,141 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+  These are the things that you have to keep in mind.
+    - Make sure that this file is also of **Java** type.
+    - You can use [*Challenge28Test.java*](https://github.com/puneeth072003/wrongsecrets/blob/Readme-updation/src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java) for reference.
```
Comment (commjoen) [inline CONTRIBUTING.md:227]:
```suggestion
Here is a unit test for reference:
```java
package org.owasp.wrongsecrets.challenges.docker;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.junit.jupiter.MockitoExtension;
import org.owasp.wrongsecrets.ScoreCard;

@ExtendWith(MockitoExtension.class)
class Challenge28Test {

    @Mock
    private ScoreCard scoreCard;

    @Test
    void rightAnswerShouldSolveChallenge() {
        var challenge = new Challenge28(scoreCard);
        Assertions.assertThat(challenge.solved("wrong answer")).isFalse();
        Assertions.assertThat(challenge.solved(challenge.spoiler().solution())).isTrue();

    }



}
```
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -99,3 +102,141 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+
+- ### Step 4: Adding explanations, reasons and hints.
+  Add the explanation for your challenge along with the hints that will help in finding the secret in this folder `wrongsecrets/src/main/resources/explanations/`.
+  Things to be noted.
+    - All the possible explanations for your challenge, included with all the hints and reasons should be provided.
+    - Every thing must be in a separate **AsciiDoc files**.
+    -  Follow this fashion in naming the file.
+        - Name the explanation in ths manner `challenge28.adoc`.
```
Comment (commjoen) [inline CONTRIBUTING.md:235]: can you add the sample here like
```asciidoc
and then the contents of the file?
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -99,3 +102,141 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+        - Reasons file as `challenge28_reason.adoc` and hints file as `challenge28_hint.adoc`.
```
Comment (commjoen) [inline CONTRIBUTING.md:236]: Same here as above

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challlenge28.java [hunk 11]
```diff
@@ -0,0 +1,69 @@
+package org.owasp.wrongsecrets.challenges.docker;
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challlenge28.java:1]: once the code-blcoks are in contributing.md you can remove this i think :)

File: src/main/resources/explanations/challenge28.adoc [hunk 12]
```diff
@@ -0,0 +1,5 @@
+=== Hello world challenge
```
Comment (commjoen) [inline src/main/resources/explanations/challenge28.adoc:1]: once the code-blcoks are in contributing.md you can remove this i think :)

File: src/main/resources/explanations/challenge28_hint.adoc [hunk 13]
```diff
@@ -0,0 +1,5 @@
+Your secret is `Hello World`
```
Comment (commjoen) [inline src/main/resources/explanations/challenge28_hint.adoc:1]: once the code-blcoks are in contributing.md you can remove this i think :)

File: src/main/resources/explanations/challenge28_reason.adoc [hunk 14]
```diff
@@ -0,0 +1,2 @@
+==== What's the purpose of this specific challenge?
```
Comment (commjoen) [inline src/main/resources/explanations/challenge28_reason.adoc:1]: once the code-blcoks are in contributing.md you can remove this i think :)

File: src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
+package org.owasp.wrongsecrets.challenges.docker;
```
Comment (commjoen) [inline src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java:1]: once the code-blcoks are in contributing.md you can remove this i think :)

**Review by commjoen — CHANGES_REQUESTED** [review]
Hi @puneeth072003 ! This is great work! Just a few more steps and i thik we are about to be there! This guide is becoming better and better! And soon everybody should be able to add cases quite easily!

File: CONTRIBUTING.md (hunk 6 above)
Comment (puneeth072003) [reply CONTRIBUTING.md:221]: @commjoen Should i remove challenge 28?

File: scripts/get_helm.sh (hunk 4 above)
Comment (puneeth072003) [reply scripts/get_helm.sh:1]: @commjoen should i unstage this folder?

File: CONTRIBUTING.md (hunk 6 above)
Comment (commjoen) [reply CONTRIBUTING.md:221]: Yes, just apply the above please

File: scripts/get_helm.sh (hunk 4 above)
Comment (puneeth072003) [reply scripts/get_helm.sh:1]: I didn't get you

Comment (commjoen) [reply scripts/get_helm.sh:1]: > @commjoen should i unstage this folder?

Yes please.

**Comment by puneeth072003** [comment]
@commjoen I'll do a fresh PR in-corparating all the changes

**Comment by bendehaan** [comment]
@puneeth072003 alright! Thanks a lot for your work so far!

**Pull request closed without merging**